e2e/test/scenarios/actions/model-actions.cy.spec.js

```diff
@@ -328,6 +333,51 @@ describe(
       cy.findByLabelText("101").should("not.exist");
       cy.findByLabelText("ID").should("be.visible");
     });
+
+    it("should show detailed form errors for constraint violations when executing model actions", () => {
+      const actionName = "Update";
+
+      cy.get("@modelId").then(modelId => {
+        cy.visit(`/model/${modelId}/detail`);
+        cy.wait("@getModel");
+      });
+
+      cy.findByRole("tablist").within(() => {
+        cy.findByText("Actions").click();
+      });
+
+      cy.wait([
+        "@fetchMetadata",
+        "@fetchMetadata",
+        "@fetchMetadata",
+        "@getModelAction",
+        "@getSearchResults",
+      ]);
+
+      createBasicActions();
+
+      cy.findByLabelText("Action list").within(() => {
+        cy.get("li").eq(1).findByText(actionName).should("be.visible");
+      });
+
+      runActionFor(actionName);
+
+      cy.wait("@getAction");
+
+      modal().within(() => {
+        cy.findByLabelText("ID").type("1");
+        cy.findByLabelText("User ID").type("999999");
+        cy.button(actionName).click();
+        cy.wait("@executeAction");
+
+        cy.findByLabelText("User ID").should("not.exist");
+        cy.findByLabelText("User ID: This User_id does not exist.").should(
+          "exist",
+        );
```

Comment:
Same comment about swapping the assertation order and removing the need to wait for `executeAction`